src/lib/tokenUtils.ts

```diff
@@ -64,6 +64,29 @@ export function formatNumber(
     .join("");
 }
 
+export function formatNumberForAdvancedDelegation(amount: string | BigInt) {
+  // Advanced delegation needs a precision up to 3 decimal places,
+  // which is bit different from the formatNumber function used everywhere else and requires for max 4 significant digits
+  const number = Number(
```

Comment:
Given you are taking both string and BigInt you could do a check like this to normalize the input

  const amountNumber = typeof amount === 'bigint' ? Number(amount) : Number(amount);
  if (isNaN(amountNumber)) {
    throw new Error("I dont have a number here");
  }

Reply:
great that you noticed it, thanks!

solving by removing the BigInt possibility, this function is used in only one place and really should only take the string